### DISCUSSION TITLE
按照品类标准模板默认添加必要实体

### DISCUSSION THREAD

**Discussion by nuomi1** [XiaoMi/ha_xiaomi_home#10] [Ideas]
目前按照 Device Spec 全量生成实体会导致数据库变得巨大（23 个设备 358 个实体），建议按照 https://iot.mi.com/v2/new/doc/introduction/solution/standard/category/switch 等模板的必要字段来创建实体，其他的实体可以通过 Configuration 文件来进行高级自定义。

**Comment by JimmyKmi** [comment]
可以在添加设备或者进入集成页面点击“配置”，勾选“隐藏非标准生成实体”以隐藏“非必要字段”生成的实体。
![screenshot-20241216-222409](https://github.com/user-attachments/assets/c7efadf1-b2f0-4daf-be5e-ffbad14b231f)

**Reply by nuomi1** [reply]
我勾选了这个还是会创建很多实体，我看看代码应该怎么调整～

**Reply by JimmyKmi** [reply]
> 我勾选了这个还是会创建很多实体，我看看代码应该怎么调整～

如果勾选了这个还是会创建非标准实体（前面带 *）的话那就是bug，欢迎提。或者看看后面能不能在添加设备的时候做成支持属性筛选添加。

**Reply by SusanPhevos** [reply]
> 我勾选了这个还是会创建很多实体，我看看代码应该怎么调整～

可以通过修改spec_filter.json达到这个目的，方法详见[README](https://github.com/XiaoMi/ha_xiaomi_home/blob/main/doc/README_zh.md#miot-spec-v2-%E8%BF%87%E6%BB%A4%E8%A7%84%E5%88%99)
